File: Java/xunit-test-patterns/src/test/java/_07_picking_test_values/_04_DerivedValues.java
package _07_picking_test_values;

import autofixture.publicinterface.Any;
import lombok.val;
import org.testng.annotations.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class _04_DerivedValues {

    @Test
    public void
    shouldCreateBackupFileNameContainingPassedHostNameAndUserName() {
        //GIVEN
        val hostName = Any.string();
        val userName = Any.string();
        val backupNamePattern = new BackupFileNamePattern();

        //WHEN
        val name = backupNamePattern.applyTo(hostName, userName);

        //THEN
        assertThat(name)
            .isEqualTo("backup_" + hostName + "_" + userName + ".zip");
    }
    //todo show extracting helper method
}

File: CSharp/XUnitTestPatterns/XUnitTestPatterns/_07_PickingTestValues/_04_DerivedValues.cs
using NUnit.Framework;
using TddXt.AnyRoot.Strings;
using static TddXt.AnyRoot.Root;

namespace XUnitTestPatterns._07_PickingTestValues
{
  public class _04_DerivedValues
  {
    [Test]
    public void ShouldCreateBackupFileNameContainingPassedHostNameAndUserName()
    {
      //GIVEN
      var hostName = Any.String();
      var userName = Any.String();
      var backupNamePattern = new BackupFileNamePattern();

      //WHEN
      var name = backupNamePattern.ApplyTo(hostName, userName);

      //THEN
      Assert.AreEqual($"backup_{hostName}_{userName}.zip", name);
    }
    //todo show extracting helper method
  }
}